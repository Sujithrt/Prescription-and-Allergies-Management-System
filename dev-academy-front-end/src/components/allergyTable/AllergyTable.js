import React from 'react';
import AddAllergyPopUp from '../addAllergyPopUp/AddAllergyPopUp';
import Container from 'react-bootstrap/Container';
import './../../pages/patientProfile/patientProfile.css';
import Alert from 'react-bootstrap/Alert';

/**
 * Displays all allergies of patient in the form of a table
 * @param {Patient} props 
 * @returns 
 */
export default function AllergyTable(props) {
    return (
        <>
            <span className="userUpdateTitle">Allergies</span><br /><br />
            <Container>
                {(props.patient.allergies && props.patient.allergies.length) ?<table className="table">
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Caused By</th>
                            <th>Reactions</th>
                            <th>Severity</th>
                            <th>Status</th>
                        </tr>
                    </thead>
                    <tbody>
                        {
                            props.patient.allergies.map(
                                allergy =>
                                    <tr key={allergy.id}>
                                        <td>{allergy.name}</td>
                                        <td>{allergy.ingredient.name}</td>
                                        <td>{allergy.reaction}</td>
                                        <td>{allergy.severity}</td>
                                        <td>{allergy.status}</td>
                                    </tr>
                            )
                        }
                    </tbody>
                </table> : <Alert variant="primary">Patient does not have any recorded allergies</Alert>}
                <AddAllergyPopUp patientDetails={props.patient} />
            </Container>
        </>
    )
}
