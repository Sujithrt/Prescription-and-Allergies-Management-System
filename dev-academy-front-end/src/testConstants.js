export const patient = {
    "id": 1001,
    "firstName": "Leanne",
    "lastName": "Graham",
    "gender": "Male",
    "age": 21,
    "phoneNumber": 9123456780,
    "email": "leanne.graham@gmail.com",
    "city": "Bangalore",
    "state": "Karnataka",
    "medicalConditions": [
        {
            "id": 1,
            "name": "Cephalalgia (headache)"
        },
        {
            "id": 3,
            "name": "Otitis externa (swimmer's ear)"
        },
        {
            "id": 9,
            "name": "Contusion (bruise)"
        }
    ],
    "allergies": [
        {
            "id": 7,
            "name": "Aspirin Allergy",
            "reaction": "Tiredness",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 4,
                "name": "Aspirin"
            },
            "encounter": {
                "id": 10,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 8,
            "name": "Paracetamol",
            "reaction": "sdfd",
            "severity": "MILD",
            "status": "ACTIVE",
            "ingredient": {
                "id": 1,
                "name": "Paracetamol"
            },
            "encounter": {
                "id": 13,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 13,
            "name": "Ibuprofin Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 20,
                "date": "2021-10-29",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 14,
            "name": "Paracetamol Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 1,
                "date": "2021-10-26",
                "patient": {
                    "id": 1003,
                    "firstName": "Clementine",
                    "lastName": "Bauch",
                    "gender": "Female",
                    "age": 56,
                    "phoneNumber": 9123456782,
                    "email": "clementine.bauch@gmail.com",
                    "city": "Mysore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 23,
                            "name": "Gestation (pregnancy) "
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        }
    ]
};

export const user = {
    "id": 1,
    "username": "John",
    "email": "john.cage@gmail.com",
    "role": "Physician",
    "firstName": "John",
    "lastName": "Cage",
    "speciality": "Cardiologist",
    "city": "Bangalore",
    "state": "Karnataka",
    "dob": "1975-06-12T18:30:00.000+00:00",
    "authorities": [
        {
            "id": 1,
            "roleCode": "USER",
            "roleDescription": "User role",
            "authority": "USER"
        },
        {
            "id": 2,
            "roleCode": "ADMIN",
            "roleDescription": "Admin role",
            "authority": "ADMIN"
        }
    ]
};

export const prescription = [
    {
        "id": {
            "encounterId": 1,
            "medicineId": 1
        },
        "encounter": {
            "id": 29,
            "date": "2021-11-07",
            "patient": {
                "id": 1009,
                "firstName": "Glenna",
                "lastName": "Reichert",
                "gender": "Female",
                "age": 65,
                "phoneNumber": 9123456788,
                "email": "glenna.reichert@gmail.com",
                "city": "Kolkata",
                "state": "West Bengal",
                "medicalConditions": [
                    {
                        "id": 6,
                        "name": "Tussis (cough)"
                    },
                    {
                        "id": 7,
                        "name": "Pyrexia (fever)"
                    }
                ]
            },
            "user": {
                "id": 1,
                "username": "John",
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "email": "john.cage@gmail.com",
                "enabled": true,
                "role": "Physician",
                "authorities": [
                    {
                        "id": 1,
                        "roleCode": "USER",
                        "roleDescription": "User role",
                        "authority": "USER"
                    },
                    {
                        "id": 2,
                        "roleCode": "ADMIN",
                        "roleDescription": "Admin role",
                        "authority": "ADMIN"
                    }
                ],
                "credentialsNonExpired": true,
                "accountNonExpired": true,
                "accountNonLocked": true
            }
        },
        "medicine": {
            "id": 1,
            "genericName": "Paracetamol",
            "brandName": "Dolo",
            "category": {
                "id": 1,
                "name": "Antipyretics"
            },
            "ingredients": [
                {
                    "id": 1,
                    "name": "Paracetamol"
                }
            ]
        },
        "dosage": "1 time a day"
    }
];

export const allergy = {
    "id": 19,
    "name": "Paracetamol Allergy",
    "reaction": "Tiredness",
    "severity": "MILD",
    "status": "ACTIVE",
    "ingredient": {
        "id": 1,
        "name": "Paracetamol"
    },
    "encounter": {
        "id": 28,
        "date": "2021-11-07",
        "patient": {
            "id": 1009,
            "firstName": "Glenna",
            "lastName": "Reichert",
            "gender": "Female",
            "age": 65,
            "phoneNumber": 9123456788,
            "email": "glenna.reichert@gmail.com",
            "city": "Kolkata",
            "state": "West Bengal",
            "medicalConditions": [
                {
                    "id": 6,
                    "name": "Tussis (cough)"
                },
                {
                    "id": 7,
                    "name": "Pyrexia (fever)"
                }
            ]
        },
        "user": {
            "id": 1,
            "username": "John",
            "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
            "email": "john.cage@gmail.com",
            "enabled": true,
            "role": "Physician",
            "authorities": [
                {
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                    "authority": "USER"
                },
                {
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                    "authority": "ADMIN"
                }
            ],
            "credentialsNonExpired": true,
            "accountNonExpired": true,
            "accountNonLocked": true
        }
    },
    "patient": {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "gender": "Female",
        "age": 65,
        "phoneNumber": 9123456788,
        "email": "glenna.reichert@gmail.com",
        "city": "Kolkata",
        "state": "West Bengal",
        "medicalConditions": [
            {
                "id": 6,
                "name": "Tussis (cough)"
            },
            {
                "id": 7,
                "name": "Pyrexia (fever)"
            }
        ]
    }
};

export const recentPatients = [
    {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-09"
    },
    {
        "id": 1008,
        "firstName": "Nicholas",
        "lastName": "Runolfsdottir",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1021,
        "firstName": "Leroy",
        "lastName": "Blair",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1027,
        "firstName": "Alanna",
        "lastName": "Cantrell",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-07"
    },
    {
        "id": 1016,
        "firstName": "Osbert",
        "lastName": "Valdez",
        "age": 32,
        "gender": "Male",
        "date": "2021-11-02"
    },
    {
        "id": 1024,
        "firstName": "Emmett",
        "lastName": "Marshall",
        "age": 45,
        "gender": "Male",
        "date": "2021-11-02"
    }
];

export const prescriptionData = [
    {
        "brandName": "Tylenol",
        "genericName": "Acetaminophen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Advil",
        "genericName": "Ibuprofen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Dolo",
        "genericName": "Paracetamol",
        "dosage": "1 time a day",
        "date": "2021-10-26"
    }
];

export const medicineList = [
    {
        "id": 1,
        "brandName": "Dolo",
        "cause": "Incompatible with patients' medical conditions"
    },
    {
        "id": 13,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 17,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 22,
        "brandName": "Betadine",
        "cause": ""
    },
    {
        "id": 25,
        "brandName": "Iodex",
        "cause": ""
    },
    {
        "id": 46,
        "brandName": "Desyrel",
        "cause": "Incompatible with patients' medical conditions"
    }
];