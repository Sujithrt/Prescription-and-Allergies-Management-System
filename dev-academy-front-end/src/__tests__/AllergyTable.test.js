import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow } from 'enzyme';
import AllergyTable from './../components/allergyTable/AllergyTable'
import { patient } from './../testConstants'

describe("AllergyTable Test", () => {
    it("AllergyTable is rendered as expected", () => {
        const allergyTable = shallow(<AllergyTable patient={patient}/>);
        expect(shallowToJson(allergyTable)).toMatchSnapshot();
    });
});

describe("Patient has no allergies", () => {
    const patientWithNoAllergy = {
        "id": 1,
        "firstName": "abc",
        "lastName": "def",
        "gender": "Male",
        "age":20,
        "phoneNumber": 1234567890,
        "email": "abc.def@gmail.com",
        "city": "Bangalore",
        "state": "Karnataka",
        "lastEncounterDate": "2021-11-03",
        "medicalConditions": [
            {
                "id": 1,
                "name": "Medical Condition 1"
            },
            {
                "id": 2,
                "name": "Medical Condition 2"
            }
        ],
        "allergies": []
    }
    it("No allergy alert displayed", () => {
        const allergyTable = shallow(<AllergyTable patient={patientWithNoAllergy}/>);
        expect(shallowToJson(allergyTable)).toMatchSnapshot();
    });
});