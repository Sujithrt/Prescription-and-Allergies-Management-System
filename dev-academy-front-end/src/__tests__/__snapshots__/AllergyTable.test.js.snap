// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllergyTable Test AllergyTable is rendered as expected 1`] = `
<Fragment>
  <span
    className="userUpdateTitle"
  >
    Allergies
  </span>
  <br />
  <br />
  <Container
    fluid={false}
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Caused By
          </th>
          <th>
            Reactions
          </th>
          <th>
            Severity
          </th>
          <th>
            Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="7"
        >
          <td>
            Aspirin Allergy
          </td>
          <td>
            Aspirin
          </td>
          <td>
            Tiredness
          </td>
          <td>
            MODERATE
          </td>
          <td>
            ACTIVE
          </td>
        </tr>
        <tr
          key="8"
        >
          <td>
            Paracetamol
          </td>
          <td>
            Paracetamol
          </td>
          <td>
            sdfd
          </td>
          <td>
            MILD
          </td>
          <td>
            ACTIVE
          </td>
        </tr>
        <tr
          key="13"
        >
          <td>
            Ibuprofin Allergy
          </td>
          <td>
            Ibuprofen
          </td>
          <td>
            Increased Perspiration
          </td>
          <td>
            MODERATE
          </td>
          <td>
            ACTIVE
          </td>
        </tr>
        <tr
          key="14"
        >
          <td>
            Paracetamol Allergy
          </td>
          <td>
            Ibuprofen
          </td>
          <td>
            Increased Perspiration
          </td>
          <td>
            MODERATE
          </td>
          <td>
            ACTIVE
          </td>
        </tr>
      </tbody>
    </table>
    <AddAllergyPopUp
      patientDetails={
        Object {
          "age": 21,
          "allergies": Array [
            Object {
              "encounter": Object {
                "date": "2021-10-26",
                "id": 10,
                "patient": Object {
                  "age": 21,
                  "city": "Bangalore",
                  "email": "leanne.graham@gmail.com",
                  "firstName": "Leanne",
                  "gender": "Male",
                  "id": 1001,
                  "lastName": "Graham",
                  "medicalConditions": Array [
                    Object {
                      "id": 1,
                      "name": "Cephalalgia (headache)",
                    },
                    Object {
                      "id": 3,
                      "name": "Otitis externa (swimmer's ear)",
                    },
                    Object {
                      "id": 9,
                      "name": "Contusion (bruise)",
                    },
                  ],
                  "phoneNumber": 9123456780,
                  "state": "Karnataka",
                },
                "user": Object {
                  "accountNonExpired": true,
                  "accountNonLocked": true,
                  "authorities": Array [
                    Object {
                      "authority": "USER",
                      "id": 1,
                      "roleCode": "USER",
                      "roleDescription": "User role",
                    },
                    Object {
                      "authority": "ADMIN",
                      "id": 2,
                      "roleCode": "ADMIN",
                      "roleDescription": "Admin role",
                    },
                  ],
                  "credentialsNonExpired": true,
                  "email": "john.cage@gmail.com",
                  "enabled": true,
                  "id": 1,
                  "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                  "role": "Physician",
                  "username": "John",
                },
              },
              "id": 7,
              "ingredient": Object {
                "id": 4,
                "name": "Aspirin",
              },
              "name": "Aspirin Allergy",
              "patient": Object {
                "age": 21,
                "city": "Bangalore",
                "email": "leanne.graham@gmail.com",
                "firstName": "Leanne",
                "gender": "Male",
                "id": 1001,
                "lastName": "Graham",
                "medicalConditions": Array [
                  Object {
                    "id": 1,
                    "name": "Cephalalgia (headache)",
                  },
                  Object {
                    "id": 3,
                    "name": "Otitis externa (swimmer's ear)",
                  },
                  Object {
                    "id": 9,
                    "name": "Contusion (bruise)",
                  },
                ],
                "phoneNumber": 9123456780,
                "state": "Karnataka",
              },
              "reaction": "Tiredness",
              "severity": "MODERATE",
              "status": "ACTIVE",
            },
            Object {
              "encounter": Object {
                "date": "2021-10-26",
                "id": 13,
                "patient": Object {
                  "age": 21,
                  "city": "Bangalore",
                  "email": "leanne.graham@gmail.com",
                  "firstName": "Leanne",
                  "gender": "Male",
                  "id": 1001,
                  "lastName": "Graham",
                  "medicalConditions": Array [
                    Object {
                      "id": 1,
                      "name": "Cephalalgia (headache)",
                    },
                    Object {
                      "id": 3,
                      "name": "Otitis externa (swimmer's ear)",
                    },
                    Object {
                      "id": 9,
                      "name": "Contusion (bruise)",
                    },
                  ],
                  "phoneNumber": 9123456780,
                  "state": "Karnataka",
                },
                "user": Object {
                  "accountNonExpired": true,
                  "accountNonLocked": true,
                  "authorities": Array [
                    Object {
                      "authority": "USER",
                      "id": 1,
                      "roleCode": "USER",
                      "roleDescription": "User role",
                    },
                    Object {
                      "authority": "ADMIN",
                      "id": 2,
                      "roleCode": "ADMIN",
                      "roleDescription": "Admin role",
                    },
                  ],
                  "credentialsNonExpired": true,
                  "email": "john.cage@gmail.com",
                  "enabled": true,
                  "id": 1,
                  "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                  "role": "Physician",
                  "username": "John",
                },
              },
              "id": 8,
              "ingredient": Object {
                "id": 1,
                "name": "Paracetamol",
              },
              "name": "Paracetamol",
              "patient": Object {
                "age": 21,
                "city": "Bangalore",
                "email": "leanne.graham@gmail.com",
                "firstName": "Leanne",
                "gender": "Male",
                "id": 1001,
                "lastName": "Graham",
                "medicalConditions": Array [
                  Object {
                    "id": 1,
                    "name": "Cephalalgia (headache)",
                  },
                  Object {
                    "id": 3,
                    "name": "Otitis externa (swimmer's ear)",
                  },
                  Object {
                    "id": 9,
                    "name": "Contusion (bruise)",
                  },
                ],
                "phoneNumber": 9123456780,
                "state": "Karnataka",
              },
              "reaction": "sdfd",
              "severity": "MILD",
              "status": "ACTIVE",
            },
            Object {
              "encounter": Object {
                "date": "2021-10-29",
                "id": 20,
                "patient": Object {
                  "age": 21,
                  "city": "Bangalore",
                  "email": "leanne.graham@gmail.com",
                  "firstName": "Leanne",
                  "gender": "Male",
                  "id": 1001,
                  "lastName": "Graham",
                  "medicalConditions": Array [
                    Object {
                      "id": 1,
                      "name": "Cephalalgia (headache)",
                    },
                    Object {
                      "id": 3,
                      "name": "Otitis externa (swimmer's ear)",
                    },
                    Object {
                      "id": 9,
                      "name": "Contusion (bruise)",
                    },
                  ],
                  "phoneNumber": 9123456780,
                  "state": "Karnataka",
                },
                "user": Object {
                  "accountNonExpired": true,
                  "accountNonLocked": true,
                  "authorities": Array [
                    Object {
                      "authority": "USER",
                      "id": 1,
                      "roleCode": "USER",
                      "roleDescription": "User role",
                    },
                    Object {
                      "authority": "ADMIN",
                      "id": 2,
                      "roleCode": "ADMIN",
                      "roleDescription": "Admin role",
                    },
                  ],
                  "credentialsNonExpired": true,
                  "email": "john.cage@gmail.com",
                  "enabled": true,
                  "id": 1,
                  "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                  "role": "Physician",
                  "username": "John",
                },
              },
              "id": 13,
              "ingredient": Object {
                "id": 3,
                "name": "Ibuprofen",
              },
              "name": "Ibuprofin Allergy",
              "patient": Object {
                "age": 21,
                "city": "Bangalore",
                "email": "leanne.graham@gmail.com",
                "firstName": "Leanne",
                "gender": "Male",
                "id": 1001,
                "lastName": "Graham",
                "medicalConditions": Array [
                  Object {
                    "id": 1,
                    "name": "Cephalalgia (headache)",
                  },
                  Object {
                    "id": 3,
                    "name": "Otitis externa (swimmer's ear)",
                  },
                  Object {
                    "id": 9,
                    "name": "Contusion (bruise)",
                  },
                ],
                "phoneNumber": 9123456780,
                "state": "Karnataka",
              },
              "reaction": "Increased Perspiration",
              "severity": "MODERATE",
              "status": "ACTIVE",
            },
            Object {
              "encounter": Object {
                "date": "2021-10-26",
                "id": 1,
                "patient": Object {
                  "age": 56,
                  "city": "Mysore",
                  "email": "clementine.bauch@gmail.com",
                  "firstName": "Clementine",
                  "gender": "Female",
                  "id": 1003,
                  "lastName": "Bauch",
                  "medicalConditions": Array [
                    Object {
                      "id": 23,
                      "name": "Gestation (pregnancy) ",
                    },
                  ],
                  "phoneNumber": 9123456782,
                  "state": "Karnataka",
                },
                "user": Object {
                  "accountNonExpired": true,
                  "accountNonLocked": true,
                  "authorities": Array [
                    Object {
                      "authority": "USER",
                      "id": 1,
                      "roleCode": "USER",
                      "roleDescription": "User role",
                    },
                    Object {
                      "authority": "ADMIN",
                      "id": 2,
                      "roleCode": "ADMIN",
                      "roleDescription": "Admin role",
                    },
                  ],
                  "credentialsNonExpired": true,
                  "email": "john.cage@gmail.com",
                  "enabled": true,
                  "id": 1,
                  "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                  "role": "Physician",
                  "username": "John",
                },
              },
              "id": 14,
              "ingredient": Object {
                "id": 3,
                "name": "Ibuprofen",
              },
              "name": "Paracetamol Allergy",
              "patient": Object {
                "age": 21,
                "city": "Bangalore",
                "email": "leanne.graham@gmail.com",
                "firstName": "Leanne",
                "gender": "Male",
                "id": 1001,
                "lastName": "Graham",
                "medicalConditions": Array [
                  Object {
                    "id": 1,
                    "name": "Cephalalgia (headache)",
                  },
                  Object {
                    "id": 3,
                    "name": "Otitis externa (swimmer's ear)",
                  },
                  Object {
                    "id": 9,
                    "name": "Contusion (bruise)",
                  },
                ],
                "phoneNumber": 9123456780,
                "state": "Karnataka",
              },
              "reaction": "Increased Perspiration",
              "severity": "MODERATE",
              "status": "ACTIVE",
            },
          ],
          "city": "Bangalore",
          "email": "leanne.graham@gmail.com",
          "firstName": "Leanne",
          "gender": "Male",
          "id": 1001,
          "lastName": "Graham",
          "medicalConditions": Array [
            Object {
              "id": 1,
              "name": "Cephalalgia (headache)",
            },
            Object {
              "id": 3,
              "name": "Otitis externa (swimmer's ear)",
            },
            Object {
              "id": 9,
              "name": "Contusion (bruise)",
            },
          ],
          "phoneNumber": 9123456780,
          "state": "Karnataka",
        }
      }
    />
  </Container>
</Fragment>
`;

exports[`Patient has no allergies No allergy alert displayed 1`] = `
<Fragment>
  <span
    className="userUpdateTitle"
  >
    Allergies
  </span>
  <br />
  <br />
  <Container
    fluid={false}
  >
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "appear": false,
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "render": [Function],
        }
      }
      variant="primary"
    >
      Patient does not have any recorded allergies
    </Alert>
    <AddAllergyPopUp
      patientDetails={
        Object {
          "age": 20,
          "allergies": Array [],
          "city": "Bangalore",
          "email": "abc.def@gmail.com",
          "firstName": "abc",
          "gender": "Male",
          "id": 1,
          "lastEncounterDate": "2021-11-03",
          "lastName": "def",
          "medicalConditions": Array [
            Object {
              "id": 1,
              "name": "Medical Condition 1",
            },
            Object {
              "id": 2,
              "name": "Medical Condition 2",
            },
          ],
          "phoneNumber": 1234567890,
          "state": "Karnataka",
        }
      }
    />
  </Container>
</Fragment>
`;
